### ISSUE TITLE
Unable to retrieve child resources

### ISSUE THREAD

**Issue by mbarnettHMX** [microsoft/AzureStorageExplorer#7474]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.31.2 (94)

### Regression From

no

### Architecture

x64

### Storage Explorer Build Number

20231003.6

### Platform

Windows

### OS Version

Windows 10 Pro, OSBuild 19045.3570

### Bug Description

We are able to connect to the resource with Azure Storage Explorer; however, when expanding Blob Containers, File Shares, Queues, or Tables we get a 'ENOTFOUND' error:

Unable to retrieve child resources

ProducerError:{
  "name": "Node Fetch Error",
  "message": "getaddrinfo ENOTFOUND <removed>.core.windows.net",
  "stack": "Error: getaddrinfo ENOTFOUND <removed>.core.windows.net\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:<removed>)",
  "cause": {
    "cause": {
      "errno": -3008,
      "code": "ENOTFOUND",
      "syscall": "getaddrinfo",
      "hostname": "<removed>..core.windows.net"
    }
  },
  "code": "ENOTFOUND",
  "errno": -3008
}

### Steps to Reproduce

I expect that this cannot be reproduced except by using our particular Storage Account Name and SAS Key.

(1) We connect with a Storage Account Name and SAS Key.
(2) connect to the resource with Azure Storage Explorer.
(3) attempt to expand Blob Containers, File Shares, Queues, or Tables.
-error-

### Actual Experience

We are able to connect to the resource with Azure Storage Explorer; however, when expanding Blob Containers, File Shares, Queues, or Tables we get a 'ENOTFOUND' error provided in a previous answer.

### Expected Experience

Our partner (Microsoft MCAPS) is trying to share files with us.

### Additional Context

none

**Comment by ChristianRazvan** [comment]
Same here:

Unable to retrieve child resources
`
ProducerError:{
  "name": "TypeError",
  "message": "{\"name\":\"TypeError\"}"
}`

**Comment by MRayermannMSFT** [comment]
@mbarnettHMX 
1. Can you double check the spelling of all of the fields in the connection string you use?
2. Is the Storage Account behind any sort of virtual network that would prevent you from accessing it from the current machine?

**Comment by RickHa79** [comment]
This is also happening to me.  Storage Explorer (SE) was working fine, then I upgraded to the latest version, (changed absolutely nothing!), and it would not connect to my Azure accounts... giving me the following error message:  
"Unable to retrieve child resources
ProducerError:{
  "name": "TypeError",
  "message": "{\"name\":\"TypeError\"}"
}"
I use SE constantly for export and import, so this is a big problem and does not speak well of Microsoft's development process.

**Comment by MRayermannMSFT** [comment]
@RickHa79 , your error appears different. Can you please open up a new issue with all the specific information about your situation? Thanks.

**Comment by RickHa79** [comment]
> @RickHa79 , your error appears different. Can you please open up a new issue with all the specific information about your situation? Thanks.

Thanks for getting back to me MRayermannMSFT.  I did a little searching and exploring with proxy settings, connection strings, and SAS settings... then finally noticed that if I clicked on one of my Storage accounts in SE, it would ask me which account I wanted to login with.  Once I signed in with my user account, the Azure accounts were recognized, and everything was back to normal.   
If there is a lesson in this, the upgrade process for SE needs to be a little more thorough: this account reconnection should be automatic... or at least tell the user what they need to do to reestablish the account credentials. 
Again, thanks for getting back to me to help!
